server/controllers/incidents.js

```diff
@@ -5,104 +5,149 @@ const Level = require("../models").Levels;
+    model: Level,
+    attributes: ["name"]
+  },
+  {
+    model: Status,
+    attributes: ["status"]
+  },
+  {
+    model: Location,
+    attributes: ["name", "centre", "country"]
+  },
+  {
+    model: User,
+    as: "assignees",
+    userAttributes,
+    through: {
+      attributes: ["assignedRole"]
+    }
+  },
+  {
+    model: User,
+    as: "reporter",
+    userAttributes,
+    through: {
+      attributes: []
+    }
+  },
+  {
+    model: User,
+    as: "witnesses",
+    userAttributes,
+    through: {
+      attributes: []
+    }
   }
-}]
+];
 
 // mapping Assignees
-const mapAssignees = (incident) => {
-return incident.map(oneIncident => {
-  oneIncident.dataValues.assignedRole = oneIncident.dataValues.assigneeIncidents.assignedRole;
+const mapAssignees = incident => {
+  return incident.map(oneIncident => {
+    oneIncident.dataValues.assignedRole =
+      oneIncident.dataValues.assigneeIncidents.assignedRole;
     delete oneIncident.dataValues.assigneeIncidents;
-    return oneIncident
-    });
-  }
+    return oneIncident;
+  });
+};
 
 module.exports = {
   // create an incident
   create(req, res) {
     let locationPromise;
-    name = req.body.name,
-    centre = req.body.centre,
-    country = req.body.country
-    return LocationService.create(name, centre, country).then(location => {
-      return location.dataValues.id;
-    }).then(locationId=> {
-      return Incident.create({
-        description: req.body.description,
-        subject: req.body.subject,
-        dateOccurred: req.body.dateOccurred,
-        statusId: req.body.statusId || 1,
-        locationId,
-        levelId: req.body.levelId
-      });
-    }).then(incident=> {
-      return User.findById(req.body.userId).then(user => {
-        return incident.addReporter(user);
-      }).then(()=> {
-        return incident;
+    (name = req.body.name),
+      (centre = req.body.centre),
+      (country = req.body.country);
+    return LocationService.create(name, centre, country)
+      .then(location => {
+        return location.dataValues.id;
+      })
+      .then(locationId => {
+        return Incident.create({
+          description: req.body.description,
+          subject: req.body.subject,
+          dateOccurred: req.body.dateOccurred,
+          statusId: req.body.statusId || 1,
+          locationId,
+          levelId: req.body.levelId
+        });
+      })
+      .then(incident => {
+        return User.findById(req.body.userId)
+          .then(user => {
+            incident.addReporter(user);
+            return incident;
+          })
+          .then(incident => {
+            let userId = req.body.id;
+            return userId
+              .map((id, index) => {
+                return User.findOrCreate({
+                  where: {
+                    id: req.body.id[index],
+                    email: req.body.email[index],
+                    imageUrl: req.body.imageUrl[index],
+                    username: req.body.username[index],
```

Comment:
Something is wrong in this section. `findOrCreate` only creates one record. On line 98, you are referring to `witnesses` which is actually one witness. Create a variable outside the map to hold the witnesses, so that at every point, push the created witness into the array of witnesses and use that in the next map.